From 341e718cf94b89a96df0b786ae0fedbd3f494016 Mon Sep 17 00:00:00 2001
From: Vlad Pelin <vlad.pelin@nxp.com>
Date: Mon, 11 Oct 2021 16:59:26 +0300
Subject: [PATCH 1/5] pkcs: update makefile and folders

split pkcs11-hse into libhse.so and libpkcs-hse.so
moved libhse and libpkcs-hse sources to own folders
updated makefile to account for split

Issue: ALB-7684
Upstream-Status: Pending 

Signed-off-by: Vlad Pelin <vlad.pelin@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 Makefile                          | 60 ++++++++++++++++++++++++-------
 {src => libhse}/hse-usr.c         |  0
 {src => libhse}/hse-usr.h         |  0
 {src => libpkcs}/pkcs11.c         |  0
 {src => libpkcs}/pkcs11.h         |  0
 {src => libpkcs}/pkcs11_context.h |  0
 {src => libpkcs}/pkcs11_crypt.c   |  0
 {src => libpkcs}/pkcs11_object.c  |  0
 {src => libpkcs}/simclist.c       |  0
 {src => libpkcs}/simclist.h       |  0
 10 files changed, 48 insertions(+), 12 deletions(-)
 rename {src => libhse}/hse-usr.c (100%)
 rename {src => libhse}/hse-usr.h (100%)
 rename {src => libpkcs}/pkcs11.c (100%)
 rename {src => libpkcs}/pkcs11.h (100%)
 rename {src => libpkcs}/pkcs11_context.h (100%)
 rename {src => libpkcs}/pkcs11_crypt.c (100%)
 rename {src => libpkcs}/pkcs11_object.c (100%)
 rename {src => libpkcs}/simclist.c (100%)
 rename {src => libpkcs}/simclist.h (100%)

diff --git a/Makefile b/Makefile
index 6d04c3f..077d642 100644
--- a/Makefile
+++ b/Makefile
@@ -3,29 +3,65 @@
 #
 
 ifeq (,$(CROSS_COMPILE))
-$(warning CROSS_COMPILE is not set, using default host compiler)
+$(error CROSS_COMPILE is not set)
 endif
-CROSS_COMPILE ?=
 
-CFLAGS ?= -shared -fPIC -Wall -fno-builtin
+CC := $(CROSS_COMPILE)gcc
+LD := $(CROSS_COMPILE)ld
+CFLAGS ?= -Wall -g
 LDFLAGS ?=
-SRCS := $(wildcard src/*.c)
 
+PKCS_LIB ?= libpkcs-hse.so
+PKCS_SDIR = libpkcs
+PKCS_ODIR = $(PKCS_SDIR)/obj
+PKCS_SRCS = $(wildcard $(PKCS_SDIR)/*.c)
+PKCS_OBJS = $(patsubst $(PKCS_SDIR)/%.c,$(PKCS_ODIR)/%.o,$(PKCS_SRCS))
+
+HSE_LIB ?= libhse.so
+HSE_MAJOR ?= 0
+HSE_MINOR ?= 9
+HSE_REV ?= 0
+HSE_PREMIUM ?= 0
+HSE_LIBVER = $(HSE_MAJOR).$(HSE_MINOR).$(HSE_REV)
 ifeq (,$(HSE_FWDIR))
-$(warning Path to HSE firmware not defined, using default location)
+$(warning Path to HSE firmware not defined, using location based on fw version; default 0.0.9.0)
 endif
-HSE_FWDIR ?= $(HOME)/HSE_FW_S32G274_0_0_9_0
+HSE_FWDIR ?= $(HOME)/HSE_FW_S32G274_$(HSE_PREMIUM)_$(HSE_MAJOR)_$(HSE_MINOR)_$(HSE_REV)
+HSE_SDIR = libhse
+HSE_ODIR = $(HSE_SDIR)/obj
+HSE_SRCS = $(wildcard $(HSE_SDIR)/*.c)
+HSE_OBJS = $(patsubst $(HSE_SDIR)/%.c,$(HSE_ODIR)/%.o,$(HSE_SRCS))
 
-INCLUDE ?= -I$(HSE_FWDIR)/interface \
+INCLUDE = -I$(HSE_FWDIR)/interface \
 		  -I$(HSE_FWDIR)/interface/inc_common \
 		  -I$(HSE_FWDIR)/interface/inc_custom \
 		  -I$(HSE_FWDIR)/interface/inc_services \
-		  -I$(HSE_FWDIR)/interface/config
+		  -I$(HSE_FWDIR)/interface/config \
+		  -I$(HSE_SDIR)
+
+all: $(PKCS_LIB)
+
+$(PKCS_LIB): $(HSE_LIB).$(HSE_LIBVER) $(PKCS_OBJS)
+	$(CC) $(CFLAGS) -shared -fPIC -L$(shell pwd) $(LDFLAGS) $(PKCS_OBJS) -o $@ -lhse
+
+$(PKCS_ODIR)/%.o: $(PKCS_SDIR)/%.c $(PKCS_ODIR)
+	$(CC) -c $(CFLAGS) $(INCLUDE) $(LDFLAGS) $< -o $@
+
+$(PKCS_ODIR):
+	mkdir -p $@
+
+$(HSE_LIB).$(HSE_LIBVER): $(HSE_OBJS)
+	$(CC) $(CFLAGS) -shared -fPIC -Wl,-soname,$(HSE_LIB).$(HSE_MAJOR) $(LDFLAGS) $(HSE_OBJS) -o $@
+	ln -s $@ $(HSE_LIB)
+
+$(HSE_ODIR)/%.o: $(HSE_SDIR)/%.c $(HSE_ODIR)
+	$(CC) -c $(CFLAGS) $(INCLUDE) $(LDFLAGS) $< -o $@
 
-all: pkcs11-hse.so
+$(HSE_ODIR):
+	mkdir -p $@
 
-pkcs11-hse.so: $(SRCS)
-	$(CROSS_COMPILE)gcc $(CFLAGS) $(INCLUDE) $(LDFLAGS) $^ -o $@
+.PHONY: clean all
 
 clean:
-	rm -f pkcs11-hse.so
+	rm -f *.so*
+	rm -rf $(PKCS_ODIR) $(HSE_ODIR)
diff --git a/src/hse-usr.c b/libhse/hse-usr.c
similarity index 100%
rename from src/hse-usr.c
rename to libhse/hse-usr.c
diff --git a/src/hse-usr.h b/libhse/hse-usr.h
similarity index 100%
rename from src/hse-usr.h
rename to libhse/hse-usr.h
diff --git a/src/pkcs11.c b/libpkcs/pkcs11.c
similarity index 100%
rename from src/pkcs11.c
rename to libpkcs/pkcs11.c
diff --git a/src/pkcs11.h b/libpkcs/pkcs11.h
similarity index 100%
rename from src/pkcs11.h
rename to libpkcs/pkcs11.h
diff --git a/src/pkcs11_context.h b/libpkcs/pkcs11_context.h
similarity index 100%
rename from src/pkcs11_context.h
rename to libpkcs/pkcs11_context.h
diff --git a/src/pkcs11_crypt.c b/libpkcs/pkcs11_crypt.c
similarity index 100%
rename from src/pkcs11_crypt.c
rename to libpkcs/pkcs11_crypt.c
diff --git a/src/pkcs11_object.c b/libpkcs/pkcs11_object.c
similarity index 100%
rename from src/pkcs11_object.c
rename to libpkcs/pkcs11_object.c
diff --git a/src/simclist.c b/libpkcs/simclist.c
similarity index 100%
rename from src/simclist.c
rename to libpkcs/simclist.c
diff --git a/src/simclist.h b/libpkcs/simclist.h
similarity index 100%
rename from src/simclist.h
rename to libpkcs/simclist.h
-- 
2.17.1

